src/cdk/table/sticky-styler.ts

```diff
@@ -178,13 +180,18 @@ export class StickyStyler {
     for (const dir of stickyDirections) {
       element.style[dir] = '';
     }
-    element.style.zIndex = this._getCalculatedZIndex(element);
 
     // If the element no longer has any more sticky directions, remove sticky positioning and
     // the sticky CSS class.
-    const hasDirection = STICKY_DIRECTIONS.some(dir => !!element.style[dir]);
-    if (!hasDirection) {
-      element.style.position = '';
+    const hasDirection = STICKY_DIRECTIONS.some(dir =>
```

Comment:
Can you expand the comment here to explain what you're doing with this check / z-index?